### ISSUE TITLE
False positive of deprecated usage after upgrad

### ISSUE THREAD

**Issue by Jack-Works** [import-js/eslint-plugin-import#2288]
## Version: 

- eslint: 7.32.0 => 8.1.0
- eslint-plugin-import: 2.24.2 => 2.25.2

## Failure code:

For type definition: (from `@mui/material`)

```ts
/**
 * @deprecated Not used internally. Use `(event: MediaQueryListEvent) => void` instead.
 */
export declare type MuiMediaQueryListListener = (event: MuiMediaQueryListEvent) => void;
export interface Options {
}
export default function useMediaQuery<Theme = unknown>(queryInput: string | ((theme: Theme) => string), options?: Options): boolean;
```

The usage of `useMediaQuery` reports: 

> Deprecated: Not used internally. Use `MediaQueryListEvent` from lib.dom.d.ts instead.eslintimport/no-deprecated

But that deprecation is actually attached to the `MuiMediaQueryList`.

**Comment by ljharb** [comment]
Indeed, it’s attached to MuiMediaQueryListListener. However, I’m not sure this is under our control - it might be the typescript parser.
